python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -158,6 +158,26 @@ def groups(*group_strs, include_sources: bool = False) -> "GroupsAssetSelection"
         check.tuple_param(group_strs, "group_strs", of_type=str)
         return GroupsAssetSelection(*group_strs, include_sources=include_sources)
 
+    @public
+    @staticmethod
+    def checks_for_assets(*assets_defs: AssetsDefinition) -> "AssetChecksForAssetKeys":
+        """Returns a selection with the asset checks that target the provided assets."""
+        return AssetChecksForAssetKeys(
+            [key for assets_def in assets_defs for key in assets_def.keys]
+        )
+
+    @public
+    @staticmethod
+    def checks(*asset_checks: AssetChecksDefinition) -> "AssetChecksForHandles":
+        """Returns a selection that includes all of the provided asset checks."""
+        return AssetChecksForHandles(
+            [
+                AssetCheckHandle(asset_key=AssetKey.from_coercible(spec.asset_key), name=spec.name)
+                for checks_def in asset_checks
+                for spec in checks_def.specs
+            ]
+        )
+
```

Comment:
Do we need separate public classes `AssetChecksForAssetKeys` and `AssetChecksForHandles`?

Would rather just export `AssetChecksCollection` or something so users don't have to know about handles.

Reply:
I think we don't consider these selection classes public. eg no one should init RequiredNeighborsAssetSelection themselves. For full hygiene we could set the return type to AssetSelection so that it's all public stuff

Reply:
Yeah I think we should change the return typehint to the publicly supported superclass type